### PR TITLE
feat(core): added findResourceById (#4397)

### PR DESCRIPTION
**Pull request by The-Alchemist**
As suggested by @Swizec .

@Swizec , could you check to see if this meets your needs/expectations?

Based on #7156 and #5719

### CODE REVIEW

File: packages/core/src/bundle.test.ts [hunk 1]
```diff
@@ -406,4 +406,23 @@ describe('Bundle tests', () => {
...
+        name: [{ given: ['John'], family: 'Doe' }],
+      };
+      const bundle: Bundle = {
+        resourceType: 'Bundle',
+        type: 'transaction',
+        entry: [{ resource: input }],
+      };
+      const result = findResourceInBundle(bundle, 'Patient', '123');
+      expect(result).toMatchObject({
+        resourceType: 'Patient',
```
Comment (mattlong) [inline packages/core/src/bundle.test.ts:494]:
```suggestion
        resourceType: 'Patient',
        id: '123',
```

File: packages/core/src/bundle.test.ts [hunk 2]
```diff
@@ -406,4 +406,23 @@ describe('Bundle tests', () => {
...
+    test('Simple resource', () => {
+      const input: Patient = {
+        resourceType: 'Patient',
+        id: '123',
+        name: [{ given: ['John'], family: 'Doe' }],
+      };
+      const bundle: Bundle = {
+        resourceType: 'Bundle',
+        type: 'transaction',
+        entry: [{ resource: input }],
```
Comment (mattlong) [inline packages/core/src/bundle.test.ts]: Could we add another couple items to `entry`? One with a different resource type but same `id`, and one with the same resource type by different `id`

Comment (The-Alchemist) [reply packages/core/src/bundle.test.ts]: good catch.  done.  added a few more tests for a few more edge cases as well
